geonode/utils.py

```diff
@@ -482,7 +487,12 @@ def default_map_config(request):
     )
 
     def _baselayer(lyr, order):
-        if settings.PROXY_BASEMAP is True:
+        # Only proxy the basemap if an ssl proxy exists for its url
+        # and the settings indicate to proxy it
+        if callable(requires_ssl_proxy) and \
+                "url" in lyr["source"] and \
+                requires_ssl_proxy(lyr["source"]["url"]) and \
+                settings.PROXY_BASEMAP is True:
```

Comment:
This should prioritize `requires_ssl_proxy` (aka `uses_proxy_route`) over `settings.PROXY_BASEMAP`, so maybe:
```py
        if (callable(requires_ssl_proxy) and \
            "url" in lyr["source"] and \
            requires_ssl_proxy(lyr["source"]["url"])) or \
            settings.PROXY_BASEMAP is True:
```